src/modules/claims/claims.service.ts

```diff
@@ -372,46 +413,96 @@ export class ClaimsService {
     return v4();
   }
 
+  validatePublishPublicClaimRequest(
+    registrationTypes: RegistrationTypes[],
+    token?: string,
+    claimType?: string
+  ) {
+    if (registrationTypes.includes(RegistrationTypes.OnChain) && !claimType) {
+      throw new Error('claimType required for on-chain registration');
+    }
+    if (registrationTypes.includes(RegistrationTypes.OffChain) && !token) {
```

Comment:
`token` isn't needed to publish not requested claim

Reply:
needs to be discussed further